utils/log/loggers/fields.go

```diff
@@ -28,14 +33,17 @@ func (o LogFields) Del(key string) {
 //AddToLogContext adds log fields to context.
 // Any info added here will be added to all logs using this context
 func AddToLogContext(ctx context.Context, key string, value interface{}) context.Context {
-	data := FromContext(ctx)
+	data := fromContext(ctx)
+	//Initialize if key doesn't exist
 	if data == nil {
-		ctx = context.WithValue(ctx, contextKey, make(LogFields))
-		data = FromContext(ctx)
+		ctx = context.WithValue(ctx, contextKey, &ProtectedLogFields{Content: make(LogFields)})
+		data = fromContext(ctx)
 	}
 	m := ctx.Value(contextKey)
-	if data, ok := m.(LogFields); ok {
-		data.Add(key, value)
+	if data, ok := m.(*ProtectedLogFields); ok {
+		data.mtx.Lock()
+		defer data.mtx.Unlock()
+		data.Content.Add(key, value)
 	}
```

Comment:
can just use `data`, without need to take `m` out from context again.  the original code seems redundant, can it be simplified to following code?

```golang
data := fromContext(ctx)
if data == nil {
    data = &ProtectedLogFields{Content: make(LogFields)}
    ctx = context.WithValue(ctx, contextKey, data)
}
data.mtx.Lock()
defer data.mtx.Unlock()
data.Content.Add(key, value)
```

Reply:
This is brilliant, I think the reason it was previously written like that was due to it needs to fetch latest copy after save, since now we are using pointer directly, this can be skipped. Great catch and thanks!